.github/CODEOWNERS

```diff
@@ -494,7 +494,11 @@ packages/kbn-repo-packages @elastic/kibana-operations
 packages/kbn-repo-path @elastic/kibana-operations
 packages/kbn-repo-source-classifier @elastic/kibana-operations
 packages/kbn-repo-source-classifier-cli @elastic/kibana-operations
+packages/kbn-reporting/common @elastic/appex-sharedux
 x-pack/examples/reporting_example @elastic/appex-sharedux
+packages/kbn-reporting/generate_csv/impl @elastic/appex-sharedux
+packages/kbn-reporting/generate_csv/mocks @elastic/appex-sharedux
+packages/kbn-reporting/generate_csv/types @elastic/appex-sharedux
```

Comment:
Based on what you found could be broken out, I think a good directory structure after the changes would be:
```
packages/kbn-reporting/common/README.md
packages/kbn-reporting/common/index.ts
packages/kbn-reporting/common/jest.config.js
packages/kbn-reporting/common/kibana.jsonc
packages/kbn-reporting/common/package.json
packages/kbn-reporting/common/src/cancellation_token.test.ts
packages/kbn-reporting/common/src/cancellation_token.ts
packages/kbn-reporting/common/src/errors.ts
packages/kbn-reporting/common/src/schema_utils.test.ts
packages/kbn-reporting/common/src/schema_utils.ts
packages/kbn-reporting/common/tsconfig.json
packages/kbn-reporting/common/types/index.d.ts

packages/kbn-generate_csv/README.mdx
packages/kbn-generate_csv/index.ts
packages/kbn-generate_csv/jest.config.js
packages/kbn-generate_csv/kibana.jsonc
packages/kbn-generate_csv/mocks/generate_csv_mocks.ts
packages/kbn-generate_csv/mocks/index.ts
packages/kbn-generate_csv/package.json
packages/kbn-generate_csv/src/generate_csv.test.ts
packages/kbn-generate_csv/src/generate_csv.ts
packages/kbn-generate_csv/src/get_export_settings.ts
packages/kbn-generate_csv/src/i18n_texts.ts
packages/kbn-generate_csv/src/max_size_builder.test.ts
packages/kbn-generate_csv/src/max_size_string_builder.ts
packages/kbn-generate_csv/tsconfig.json
packages/kbn-generate_csv/types/index.d.ts

```

Reply:
I hit a circular dependency type issue in the CI because @kbn/reporting-plugin was referencing @kbn/generate-csv and vice versa. I'm going to add the types package to resolve this issue but otherwise I think I am following your above comment thanks! 

```

Check Types | 6s
-- | --
  | info Building TypeScript projects to check types (For visible, though excessive, progress info you can pass --verbose)
  | info starting [tsc] > node_modules/typescript/bin/tsc -b tsconfig.refs.json --pretty
  | proc [tsc] error TS6202: Project references may not form a circular graph. Cycle detected: /dev/shm/builds/kb-n2-16-spot-824d85782f65aa1b/elastic/kibana-pull-request/kibana/tsconfig.refs.json
  | proc [tsc] /dev/shm/builds/kb-n2-16-spot-824d85782f65aa1b/elastic/kibana-pull-request/kibana/packages/kbn-generate-csv/tsconfig.type_check.json
  | proc [tsc] /dev/shm/builds/kb-n2-16-spot-824d85782f65aa1b/elastic/kibana-pull-request/kibana/x-pack/plugins/reporting/tsconfig.type_check.json
  | proc [tsc]
  | proc [tsc]
  | proc [tsc] Found 1 error.
  | proc [tsc]
  | ERROR Unable to build TS project refs
  | 🚨 Error: The command exited with status 1

```